pkg/collect/incluster-cert-collector.go

```diff
@@ -0,0 +1,124 @@
+package collect
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"log"
+	"time"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type CollectInclusterCertificate struct {
+	Collector    *troubleshootv1beta2.InclusterCertificate
+	BundlePath   string
+	Namespace    string
+	ClientConfig *rest.Config
+	Client       kubernetes.Interface
+	Context      context.Context
+	RBACErrors
+}
+
+type Secret struct {
+	Name      string `json:"secret Name,omitempty"`
+	Namespace string `json:"secret Namespace,omitempty"`
+}
+
+// Certificate Struct
+type ParsedCertificate struct {
+	CertName         string    `json:"Certificate Name"`
+	DNSNames         []string  `json:"DNS Names"`
+	IssuerCommonName string    `json:"Issuer"`
+	Organizations    []string  `json:"Issuer Organizations"`
+	CertDate         time.Time `json:"Certificate Expiration Date"`
+	IsValid          bool      `json:"IsValid"`
+	SecretInfo       Secret    `json:"Secret"`
+}
+
+func (c *CollectInclusterCertificate) Title() string {
+	return getCollectorName(c)
+}
+
+func (c *CollectInclusterCertificate) IsExcluded() (bool, error) {
+	return isExcluded(c.Collector.Exclude)
+}
+
+func (c *CollectInclusterCertificate) Collect(progressChan chan<- interface{}) (CollectorResult, error) {
+
+	output := NewResult()
+	// Json object initilization - start
+	var certInfo []ParsedCertificate
+	var certJson = []byte("[]")
+	errJson := json.Unmarshal(certJson, &certInfo)
+	if errJson != nil {
+		return nil, errors.Wrap(errJson, "failed to umarshal Json")
+	} // Json object initilization - end
+
+	certificate := SecretCertCollector(c.Collector.Name, c.Client)
+
+	results := certificate
+
+	filePath := "certificates/" + c.Collector.Name + ".json"
+
+	output.SaveResult(c.BundlePath, filePath, bytes.NewBuffer(results))
+
+	return output, nil
+}
+
+func SecretCertCollector(secretName string, client kubernetes.Interface) []byte {
+
+	currentTime := time.Now()
+	var certInfo []ParsedCertificate
+	var certJson = []byte("[]")
+	err := json.Unmarshal(certJson, &certInfo)
+	if err != nil {
+		log.Println(err)
+	}
```

Comment:
Are these lines needed?

Reply:
I believe they are; I was getting a json error before; I believe that it was due to attempting to write to an empty json object but cannot remember exactly now. This is what worked for me but perhaps there is a better way?

Reply:
I think so but maybe we should walk through it together.